File: Sample/HelloWorld/Server/src/Script/PyScript/Action/Action100.py
"""100_Hello World"""
import ReferenceLib
from action import *
from ZyGames.Framework.Cache.Generic import *

class UrlParam(HttpParam):
    def __init__(self):
        HttpParam.__init__(self)


class ActionResult(DataResult):
    def __init__(self):
        DataResult.__init__(self)
        self._content = ''


def getUrlElement(httpGet, parent):
    urlParam = UrlParam()
    urlParam.Result = True
    return urlParam

def takeAction(urlParam, parent):
    actionResult = ActionResult()
    actionResult._content = 'Hello World for Python!'
    return actionResult

def buildPacket(writer, urlParam, actionResult):
    writer.PushIntoStack(actionResult._content)

    return True

File: Sample/HelloWorld/Server/src/Script/LuaScript/Action100.lua
Action100 ={}

function Action100:getUrlElement(httpGet, parent)
    local urlParam = {}
    urlParam.Result = true
    return urlParam
end

function Action100:takeAction(urlParam, parent)
	CPrint("use lua do action...")
	local actionResult = {}
	actionResult.Result = true
	actionResult._content = "Hello World for Lua!"
    return actionResult
end


function Action100:buildPacket(writer, urlParam, actionResult)
	PushIntoStack(writer, actionResult._content)
    return true
end

